src/diagram.cpp

```diff
@@ -626,14 +628,14 @@ void TreeDiagram::drawBoxes(TextStream &t,Image *image,
           }
           else
           {
-            xf = di->xPos()/static_cast<float>(gridWidth);
+            xf = di->xfPos()/static_cast<float>(gridWidth);
```

Comment:
by using xfPos() the float cast of gridWidth could be removed, or does that then still give a compiler warning?

Reply:
Interesting thought.
I did a quick test and the warning does not re-appear, makes code easier to read.
I'm gone work on it.

Reply:
I corrected the mentioned conversion.

- Should for the `static_cast<float>(superRows)` a separate variable be created like:
   ```
   float superRowsF = static_cast<float>(superRows)
   ```
   so that the construct
   ```
   yf = static_cast<float>(superRows)-1-di->yfPos()/gridHeight;
   ```
   can be rewritten as:
   ```
   yf = superRowsF-1-di->yfPos()/gridHeight;
   ```